app/controllers/users_controller.rb

```diff
@@ -4,9 +4,14 @@ def new
 
   def create
     user = User.create(user_params)
-    flash[:success] = "You are now registered and logged in!"
-    session[:user_id] = user.id
-    redirect_to '/profile'
+    if user.save
+      flash[:success] = "You are now registered and logged in!"
+      session[:user_id] = user.id
+      redirect_to '/profile'
+    else
+      flash[:error] = user.errors.full_messages.to_sentence
+      redirect_to '/register'
+    end
   end
```

Comment:
Good use of if/else here for login credentials. Good use of sad path flash message